### PR TITLE
Use `git clone` to get source

### LINKED ISSUE #550: `git clone` should be used for getting source
/kind bug

**What steps did you take and what happened:**
Trying to follow the steps in the [developers guide](https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/main/docs/book/src/developers/development.md#get-the-source) to get the source but facing errors.

```
$ go get -d sigs.k8s.io/cluster-api-provider-gcp
go: downloading sigs.k8s.io/cluster-api-provider-gcp v1.0.1
go get: sigs.k8s.io/cluster-api-provider-gcp@v1.0.1 requires
        sigs.k8s.io/cluster-api/test@v1.0.4 requires
        sigs.k8s.io/cluster-api@v0.0.0-00010101000000-000000000000: invalid version: unknown revision 000000000000
```

**What did you expect to happen:**
Expected the command to work.

**Anything else you would like to add:**
Using `git clone` will solve this issue.

**Environment:**
- go version: go1.17.6 linux/amd64
- OS (e.g. from `/etc/os-release`): ubuntu 20.04

### PR DESCRIPTION
**Pull request by jayesh-srivastava**
**What type of PR is this?**
/kind cleanup
/kind documentation

**What this PR does / why we need it**:
Existing method of getting source is failing, replaced it with `git clone` method.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #550 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:
```
NONE
```

### CODE REVIEW

**Comment by sbueringer** [comment]
/lgtm

I think the link check might have to be fixed first (kustomize dropped their doc in the repo, I think this should be changed to a link to somewhere here: https://kustomize.io/

**Comment by cpanato** [comment]
@jayesh-srivastava can you check the ci failure?

**Comment by jayesh-srivastava** [comment]
@cpanato @sbueringer the specific link for installing kustomize on linux is broken. Should I fix it in this PR only or a separate one?

**Comment by cpanato** [comment]
> @cpanato @sbueringer the specific link for installing kustomize on linux is broken. Should I fix it in this PR only or a separate one?

fix in this one

File: docs/book/src/developers/development.md [hunk 1]
```diff
@@ -19,7 +19,7 @@
 5. Install [Kustomize][kustomize]
    - `brew install kustomize` on macOS.
    - [install instructions](https://kubectl.docs.kubernetes.io/installation/kustomize/) on Windows + WSL2.
-   - [install instructions][kustomizelinux] on Linux
+   - [install instructions](https://kubectl.docs.kubernetes.io/installation/kustomize/) on Linux
```
Comment (sbueringer) [inline docs/book/src/developers/development.md:22]: Should we just conflate this into one line given that the link is now the same?

Should we also drop the now unused kustomizelinux link at the bottom of the doc?

Comment (jayesh-srivastava) [reply docs/book/src/developers/development.md:22]: Sounds good to me. Will do.

**Comment by sbueringer** [comment]
Thank you!

/lgtm

**Comment by cpanato** [comment]
thanks
/approve

**Pull request merged**